### PR TITLE
Add support for float type in mapstriface

### PR DESCRIPTION
**Pull request by ruflin**
Extracted from https://github.com/elastic/beats/pull/6807

### CODE REVIEW

File: libbeat/common/schema/mapstriface/mapstriface.go [hunk 1]
```diff
@@ -193,9 +193,43 @@ func toInteger(key string, data map[string]interface{}) (interface{}, error) {
...
+	return schema.SetOptions(schema.Conv{Key: key, Func: toFloat}, opts)
+}
+
+func toFloat(key string, data map[string]interface{}) (interface{}, error) {
+	emptyIface, exists := data[key]
+	if !exists {
+		return 0, fmt.Errorf("Key %s not found", key)
+	}
+	switch emptyIface.(type) {
+	case float64:
```
Comment (andrewkroh) [inline libbeat/common/schema/mapstriface/mapstriface.go]: Should `string` be accepted?

Comment (urso) [reply libbeat/common/schema/mapstriface/mapstriface.go]: +1 on parsing string via `strconv.ParseFloat`.

File: libbeat/common/schema/mapstriface/mapstriface.go [hunk 2]
```diff
@@ -193,9 +193,43 @@ func toInteger(key string, data map[string]interface{}) (interface{}, error) {
...
+// Float creates a Conv object for converting floats. Acceptable input
+// types are int64, int, and float64.
+func Float(key string, opts ...schema.SchemaOption) schema.Conv {
+	return schema.SetOptions(schema.Conv{Key: key, Func: toFloat}, opts)
+}
+
+func toFloat(key string, data map[string]interface{}) (interface{}, error) {
+	emptyIface, exists := data[key]
+	if !exists {
+		return 0, fmt.Errorf("Key %s not found", key)
```
[the remark this answers is not in the document]
Comment (ruflin) [reply libbeat/common/schema/mapstriface/mapstriface.go]: changed

File: libbeat/common/schema/mapstriface/mapstriface.go (hunk 1 above)
Comment (ruflin) [reply libbeat/common/schema/mapstriface/mapstriface.go]: So far we don't need it and it's also not supported in the Int type. I would add it when we need it.

Comment (andrewkroh) [reply libbeat/common/schema/mapstriface/mapstriface.go]: That’s fine by me to omit it for now. It’s an easy add at any point.

**Pull request merged**